src/notifier.js

```
@@ -64,30 +64,34 @@ function sendRecoveryEmail(displayId, addresses) {
 function prepareAndSendEmail(template, displayId, recipients) {
   const subject = template.subject.replace('DISPLAYID', displayId);
   const text = template.body.replace(/DISPLAYID/g, displayId);
+  const promises = [];
+
+  recipients.forEach(recipient=>{
+    const data = {
+      from: SENDER_ADDRESS,
+      fromName: SENDER_NAME,
+      recipients: recipient,
+      subject,
+      text: text.replace("EMAIL", recipient)
+    };
+
+    promises.push(send(data));
+  });
 
-  const data = {
-    from: SENDER_ADDRESS,
-    fromName: SENDER_NAME,
-    recipients,
-    subject,
-    text
-  };
-
-  return send(data);
+  return Promise.all(promises);
```

Is it OK to fail fast here? Some recipients might not receive the notification if there's a problem in one of them

---

Interesting point, as any rejection will automatically cause return a failed promise. I wonder if non-rejected promises are guaranteed to finish even in that case, or they are not completed.

---

All of the promises are scheduled when the array is constructed.
Promise.all will fail fast on the first rejected promise, but I don't think it is going to cancel execution of all of the remaining unresolved promises.

---

Confirmed.

---

Thanks. 👍